cloud-gov/cg-ui
# Resolve 409 org selection component

## Changes proposed in this pull request:
- [x] Break out into separate React components. Add prop for single vs multiple orgs. 
- [x] Change display of single org vs multiple orgs.

## How to review these changes locally:
1. Check out and run this branch.
2. Navigate to http://localhost:3000/prototype/design-guide.
3. Head over to **page.tsx**, line 42. To display only one organization, set the prop to `{true}`.

```javascript
<OrgPicker single={false} />
```

- [x] Collapse the component and focus back on toggle button when: 1) clicking outside of the component, 2) tabbing outside, or 3) pressing ESC. 
~~⚠️ Didn't want to block this PR review. We can create a ticket to add this behavior. I can work on it when I get back next week.~~

### Related issues
Resolves #409 
Blocks #408, #410 
Needed for #518 

### Submitter checklist


## Security considerations

Not sure if this is a security consideration, but confirmed with the engineers users will always be part of at least **one** organization. I did not add a state for **no** organization.

## Design considerations
Discussed the following with Ethan, who gave a 👍🏾 
* Using a cross-browser html entity for the **View all organizations** arrow, for now, instead of an icon. Looking into handling USWDS icons with the `use` option, so icon-size and text-color can be applied. 
* Using `scrollbar-width: thin` for overflow. This doesn’t work on Safari or Android browsers. Fall back to browser default.
* A11Y:  We'll keep an eye out for touch target, making sure they're large enough on mobile org links.
